optashift-employee-rostering-server/src/main/java/org/optaplanner/openshift/employeerostering/server/employee/EmployeeRestServiceImpl.java

```diff
@@ -94,17 +96,26 @@ protected void validateTenantIdParameter(Integer tenantId, Employee employee) {
 
     @Override
     @Transactional
-    public Long addEmployeeAvailability(Integer tenantId, EmployeeAvailabilityView employeeAvailabilityView) {
+    public EmployeeAvailabilityView addEmployeeAvailability(Integer tenantId, EmployeeAvailabilityView employeeAvailabilityView) {
         EmployeeAvailability employeeAvailability = convertFromView(tenantId, employeeAvailabilityView);
         entityManager.persist(employeeAvailability);
-        return employeeAvailability.getId();
+        return new EmployeeAvailabilityView(employeeAvailability);
     }
 
     @Override
     @Transactional
-    public void updateEmployeeAvailability(Integer tenantId, EmployeeAvailabilityView employeeAvailabilityView) {
+    public EmployeeAvailabilityView updateEmployeeAvailability(Integer tenantId, EmployeeAvailabilityView employeeAvailabilityView) {
         EmployeeAvailability employeeAvailability = convertFromView(tenantId, employeeAvailabilityView);
         entityManager.merge(employeeAvailability);
```

Comment:
the merge method returns another instance (usually).
This should should be

employeeAvailability = entityManager.merge(employeeAvailability);

Reply:
Version isn't updated in the merge, since a transaction need to be committed for hibernate to increment version.